tests/test_makefile.sh

```diff
@@ -10,43 +10,228 @@ function cleanup() {
 }
 trap cleanup EXIT
 
+function make_check_clean_target() {
+    make -s DESTDIR="$TMPDIR/install" DEFINES_FILE="$SRCDIR"/mender_grubenv_defines.example $1
+    if find "$TMPDIR" | egrep -v "^$TMPDIR\$|\.log$" | egrep '^.+$'; then
+        echo "$1 target did not clean up!"
+        return 1
+    fi
+}
+
 cd "$SRCDIR"
 
 ################################################################################
-# Normal install.
+
+make -s DESTDIR="$TMPDIR/install" DEFINES_FILE="$SRCDIR"/mender_grubenv_defines.example install-standalone-boot-files
+
+find "$TMPDIR/install" -type f -o -type l | sed -e "s,^$TMPDIR/install/,," | sort > "$TMPDIR/actual.log"
+cat > "$TMPDIR/expected.log" <<EOF
+boot/efi/EFI/BOOT/grub.cfg
+boot/efi/grub-mender-grubenv/mender_grubenv1/env
+boot/efi/grub-mender-grubenv/mender_grubenv1/lock
+boot/efi/grub-mender-grubenv/mender_grubenv1/lock.sha256sum
+boot/efi/grub-mender-grubenv/mender_grubenv2/env
+boot/efi/grub-mender-grubenv/mender_grubenv2/lock
+boot/efi/grub-mender-grubenv/mender_grubenv2/lock.sha256sum
+EOF
+diff -u "$TMPDIR/expected.log" "$TMPDIR/actual.log"
+
+make_check_clean_target uninstall-standalone-boot-files
+
+rm -rf "$TMPDIR"/*
```

Comment:
despite this being a test, and TMPDIR not being empty most probably, I would consider:
```bash
[ "${TMPDIR}" != "" ] && rm -rf "$TMPDIR"/*
```
I know, paranoia mode:on.

Reply:
Hehe, probably a good point. I'll do it in a simpler way and simply exit early if it fails to be set at the top.